common/modules/buffer.jsm

```diff
@@ -731,9 +731,14 @@ var Buffer = Module("Buffer", {
         }
 
         let link = DOM("link[href][rev=canonical], \
+                        link[href][rel=shorturl], \
                         link[href][rel=shortlink]", doc)
                        .filter(sane)
                        .attr("href");
+
+        if (!link)
+            link = DOM("link[href][rel=canonical]", doc).filter(sane).attr("href");
```

Comment:
I don't think I agree. The other link types are meant to provide short versions of a URL that lead to the same page. The `rel=canonical` links are sometimes meant to point to a better URL for the current page (which might be longer), or to another page entirely, which might be a non-paginated version of the article, or the same content on a different part of the site. I don't think this is the expected behavior for `y`.